decidim-core/spec/features/authentication_spec.rb

```diff
@@ -11,22 +11,91 @@
   end
 
   describe "Sign Up" do
-    it "creates a new User" do
-      find(".sign-up-link").click
+    context "using email and password" do
+      it "creates a new User" do
+        find(".sign-up-link").click
 
-      within ".new_user" do
-        fill_in :user_email, with: "user@example.org"
-        fill_in :user_name, with: "Responsible Citizen"
-        fill_in :user_password, with: "123456"
-        fill_in :user_password_confirmation, with: "123456"
-        check :user_tos_agreement
-        find("*[type=submit]").click
+        within ".new_user" do
+          fill_in :user_email, with: "user@example.org"
+          fill_in :user_name, with: "Responsible Citizen"
+          fill_in :user_password, with: "123456"
+          fill_in :user_password_confirmation, with: "123456"
+          check :user_tos_agreement
+          find("*[type=submit]").click
+        end
+
+        expect(page).to have_content("confirmation link")
+        expect(emails.count).to eq(1)
+        expect(last_user.email).to eq("user@example.org")
+        expect(last_user.organization).to eq(organization)
+      end
+    end
+
+    context "using facebook" do
+      let(:verified) { true }
+      let(:omniauth_hash) {
+        OmniAuth::AuthHash.new({
+          provider: 'facebook',
+          uid: '123545',
+          info: {
+            email: "user@from-facebook.com",
+            name: "Facebook User",
+            verified: verified
+          }
+        })
+      }
+
+      before :each do
+        OmniAuth.config.test_mode = true
+        OmniAuth.config.mock_auth[:facebook] = omniauth_hash
       end
 
-      expect(page).to have_content("confirmation link")
-      expect(emails.count).to eq(1)
-      expect(last_user.email).to eq("user@example.org")
-      expect(last_user.organization).to eq(organization)
+      after :each do
+        OmniAuth.config.test_mode = false
+        OmniAuth.config.mock_auth[:facebook] = nil         
+      end
+
+      context "when the user has confirmed the email in facebook" do
+        it "creates a new User without sending confirmation instructions" do
+          find(".sign-up-link").click
+
+          click_link "Sign in with Facebook"   
+
+          expect(page).to have_content("Successfully authenticated from Facebook account.")
+          expect(page).to have_content(last_user.name)
+          expect(last_user.email).to eq("user@from-facebook.com")
```

Comment:
We shouldn't assert other things than the ones the user is seeing in a feature spec